destination/destination_integration_test.go

```diff
@@ -0,0 +1,252 @@
+	"github.com/google/uuid"
+	"github.com/matryer/is"
+)
+
+type ResultConfig struct {
+	Payload   Data   `json:"payload"`
+	MetaData  string `json:"meta_data"`
+	Position  string `json:"position"`
+	Key       string `json:"key"`
+	Operation string `json:"operation"`
+}
+
+type Data struct {
+	Before string `json:"before"`
+	After  string `json:"after"`
+}
+
+func TestDestination_SuccessfulMessageSend(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+
+	metadata := sdk.Metadata{}
+	destination := NewDestination()
+	defer func() {
+		err := destination.Teardown(ctx)
+		is.NoErr(err)
+	}()
+
+	messageBody := "Test message body"
+	record := sdk.Util.Source.NewRecordCreate(
+		sdk.Position("111111"),
+		metadata,
+		sdk.RawData("1111111"),
+		sdk.RawData(messageBody),
+	)
+
+	client, url, cfg, err := prepareIntegrationTest(t)
+	is.NoErr(err)
+
+	err = destination.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = destination.Open(ctx)
+	is.NoErr(err)
+
+	ret, err := destination.Write(ctx, []sdk.Record{record})
+	is.NoErr(err)
+
+	is.Equal(ret, 1)
+	time.Sleep(5 * time.Second)
+
+	message, err := client.ReceiveMessage(
+		context.Background(),
+		&sqs.ReceiveMessageInput{
+			QueueUrl: url.QueueUrl,
+		},
+	)
+
+	is.NoErr(err)
+	is.Equal(len(message.Messages), 1)
+
+	var result ResultConfig
+	err = json.Unmarshal([]byte(*message.Messages[0].Body), &result)
+	is.NoErr(err)
+	bodyDecoded, err := base64.StdEncoding.DecodeString(result.Payload.After)
+
+	is.NoErr(err)
+	is.Equal(string(bodyDecoded), messageBody)
+}
+
+func TestDestination_FailBadRecord(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+
+	metadata := sdk.Metadata{}
+	destination := NewDestination()
+	defer func() {
+		err := destination.Teardown(ctx)
+		is.NoErr(err)
+	}()
+
+	messageBody := "Test message body"
+	record := sdk.Util.Source.NewRecordCreate(
+		sdk.Position(""),
+		metadata,
+		sdk.RawData(""),
+		sdk.RawData(messageBody),
+	)
+
+	_, _, cfg, err := prepareIntegrationTest(t)
+	is.NoErr(err)
+
+	err = destination.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = destination.Open(ctx)
+	is.NoErr(err)
+
+	_, err = destination.Write(ctx, []sdk.Record{record})
+	is.True(strings.Contains(err.Error(), "AWS.SimpleQueueService.InvalidBatchEntryId"))
+}
+
+func TestDestination_FailNonExistentQueue(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+
+	destination := NewDestination()
+	defer func() {
+		err := destination.Teardown(ctx)
+		is.NoErr(err)
+	}()
+
+	_, _, cfg, err := prepareIntegrationTest(t)
+	is.NoErr(err)
+
+	cfg[ConfigKeyAWSQueue] = ""
+
+	err = destination.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = destination.Open(ctx)
+	is.True(strings.Contains(err.Error(), "AWS.SimpleQueueService.NonExistentQueue"))
+}
+
+func prepareIntegrationTest(t *testing.T) (*sqs.Client, *sqs.GetQueueUrlOutput, map[string]string, error) {
+	cfg, err := parseIntegrationConfig()
+	if err != nil {
+		t.Skip(err)
+	}
```

Comment:
Nitpick: I believe we shouldn't be skipping integration tests if the configuration is not OK. If we want the tests to run, then they need to be run. Whether we want them to run or not is a different thing (we can control that through a flag).

Reply:
Saw this in some other connectors, but I'll remove it! :)